core/lib/page-functions.js

```diff
@@ -174,7 +174,7 @@ function getMaxTextureSize() {
     const gl = canvas.getContext('webgl');
     if (!gl) throw new Error('no webgl');
     const maxTextureSize = gl.getParameter(gl.MAX_TEXTURE_SIZE);
-    return maxTextureSize;
+    return maxTextureSize - 1;
```

Comment:
This doesn't seem right... the value from GL parameters should be a valid texture dimension. Do you have any insight on why this is needed?

Reply:
> Do you have any insight on why this is needed?

Only from testing. This single change really was the difference maker for some reason.

I was playing around with this script if it helps:

```js
import puppeteer from 'puppeteer';

const browser = await puppeteer.launch({
  headless: false,
  ignoreDefaultArgs: ['--enable-automation'],
});
const page = await browser.newPage();
await page.emulate(puppeteer.devices['Moto G4']);
const session = await page.target().createCDPSession();

await page.goto('http://localhost:10200/max-texture-size.html');

await session.send('Emulation.setDeviceMetricsOverride', {
  mobile: true,
  deviceScaleFactor: 1,
  height: Number(process.argv[2]),
  width: 0, // Leave width unchanged
});

const result = await session.send('Page.captureScreenshot', {
  format: 'webp',
  quality: 30,
})

console.log(result.data.substring(0, 50));

await page.close();
await browser.close();
```

Reply:
Can you verify if uploading a texture via an image of that exact size also fails? Here's some quick code to get you started on interfacing with gl: https://stackoverflow.com/q/50044962

just need an image of that size to test with

Reply:
Yeah, will verify before merging this.

Reply:
Max texture size on my macbook: 16384
Max texture size on my linux cloudtop: 8192

On both machines, the maximum height I could get for a screenshot was 16383 (i.e. macbook max texture size - 1). It seems like this max screenshot size is not tied directly to the max gl texture size.